src/configupdater/configupdater.py

```diff
@@ -87,592 +44,8 @@ def __init__(self):
-
-        Returns:
-            self for chaining
-        """
-        comment = Comment(self._container)
-        if not text.startswith(comment_prefix):
-            text = "{} {}".format(comment_prefix, text)
-        if not text.endswith("\n"):
-            text = "{}{}".format(text, "\n")
-        comment.add_line(text)
-        self._container.structure.insert(self._idx, comment)
-        self._idx += 1
-        return self
-
-    def section(self: BB, section) -> BB:
-        """Creates a section block
-
-        Args:
-            section (str or :class:`Section`): name of section or object
-
-        Returns:
-            self for chaining
-        """
-        if not isinstance(self._container, ConfigUpdater):
-            raise ValueError("Sections can only be added at section level!")
-
-        container: ConfigUpdater = self._container
-
-        if isinstance(section, str):
-            # create a new section
-            section = Section(section, container=container)
-        elif not isinstance(section, Section):
-            raise ValueError("Parameter must be a string or Section type!")
-
-        if container.has_section(section.name):
-            raise DuplicateSectionError(section.name)
-
-        self._container.structure.insert(self._idx, section)
-        self._idx += 1
-        return self
-
-    def space(self: BB, newlines=1) -> BB:
-        """Creates a vertical space of newlines
-
-        Args:
-            newlines (int): number of empty lines
-
-        Returns:
-            self for chaining
-        """
-        space = Space(container=self._container)
-        for _ in range(newlines):
-            space.add_line("\n")
-        self._container.structure.insert(self._idx, space)
-        self._idx += 1
-        return self
-
-    def option(self: BB, key, value=None, **kwargs) -> BB:
-        """Creates a new option inside a section
-
-        Args:
-            key (str): key of the option
-            value (str or None): value of the option
-            **kwargs: are passed to the constructor of :class:`Option`
-
-        Returns:
-            self for chaining
-        """
-        if not isinstance(self._container, Section):
-            raise ValueError("Options can only be added inside a section!")
-        option = Option(key, value, container=self._container, **kwargs)
-        if option.key in self._container.options():
-            raise DuplicateOptionError(self._container.name, option.key)
-        option.value = value
-        self._container.structure.insert(self._idx, option)
-        self._idx += 1
-        return self
-
-
-class ConfigUpdater(Container[ConfigContent], MutableMapping[str, Section]):
-    """Parser for updating configuration files.
+class ConfigUpdater(Document):
+    """Tool to parse and modify existing ``cfg`` files.
 
     ConfigUpdater follows the API of ConfigParser with some differences:
       * inline comments are treated as part of a key's value,
```

Comment:
Should here a new-line be prepended just like in the following enumeration?

Reply:
Not sure if I understood this comment

Reply:
My bad :-) I just wondered if there is a difference in the rendering if an enumeration in Markdown starting with * is rendered differently if there is a new line in before the first enumeration item. So rather
```
ConfigUpdater follows the API of ConfigParser with some differences:      

* inline comments are treated as part of a key's value,
```
Not sure about this point but in most other enumerations we have a newline.

Reply:
I think it will depend on the parser you use for the docs. I think the `rst` Parser can handle it fine (we have to check in the RTD).
Maybe some Markdown parsers don't :(

Reply:
I have added the new line in the latest commits.